ports/psoc6/machine_sdcard.c

```diff
@@ -0,0 +1,297 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020-2021 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// std includes
+#include <string.h>
+
+// micropython includes
+#include "py/runtime.h"
+#include "extmod/vfs.h"
+#include "modpsoc6.h"
+#include "mplogger.h"
+#include "mphalport.h"
+
+// port-specific includes
+#include "modmachine.h"
+#include "machine_pin_phy.h"
+#include "mplogger.h"
+
+// MTB includes
+#include "cyhal.h"
+#include "cybsp.h"
+
+#define SDHC_DEFAULT_BUS_WIDTH                  (4U)
+#define SDHC_BLOCK_SIZE                         (512UL)
+#define SDHC_NUM_OF_SLOTS                       (2)
```

Comment:
Should this be here? This is true only for proto or? 
I think number of slots is variable as per hardware and this should end up somewhere in boards config?

Reply:
Yes I will move the defines to board config.